ballerina/http_interceptors.bal

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# The HTTP request interceptor service object  
+public type RequestInterceptor distinct service object {
+
+};
+
+# The HTTP request error interceptor service object  
+public type RequestErrorInterceptor distinct service object {
+
+};
+
+# The return values from an interceptor service function  
+public type NextService RequestInterceptor|RequestErrorInterceptor|Service;
```

Comment:
I am just wondering if it should be,

```ballerina
public type NextService RequestInterceptor|Service;
```

Reply:
As per the current proposal `next` could only control the flow of RequestInterceptor|Service interceptors.

Reply:
Fixed in [`c101943`](https://github.com/ballerina-platform/module-ballerina-http/pull/713/commits/c1019434762e8544d2380b9e543ceb9b74121a23)